src/ipc/ipc4/helper.c

```diff
@@ -250,6 +256,12 @@ __cold static int ipc4_create_pipeline_payload_decode(char *data)
 	tr_info(&ipc_tr, "payload size %u array %u: %#x", hdr->payload_words, hdr->data_obj_array,
 		*((uint32_t *)hdr));
 
+	if (ALIGN_UP(size, CONFIG_DCACHE_LINE_SIZE) > hdr_cache_size)
+		dcache_invalidate_region((__sparse_force void __sparse_cache *)
+					 ((char *)MAILBOX_HOSTBOX_BASE + hdr_cache_size),
+					 ALIGN_UP(size, CONFIG_DCACHE_LINE_SIZE) -
+					 hdr_cache_size);
```

Comment:
it should be safe to just use whatever pointers and sizes you have, cache management functions adjust for cache lines internally